### ISSUE TITLE
hie fails to compile mutually recursive modules

### ISSUE THREAD

**Issue by PanAeon** [haskell/haskell-ide-engine#1134]
With mutually recursive modules, e.g.
```

# ------------- ModuleA.hs ---------------------------
module ModuleA where

import {-# SOURCE #-} ModuleB

newtype A = A Int   
# -------------ModuleB.hs ---------------------------

module ModuleB where

import ModuleA

foo :: Int
foo = 34  

# ------------ModuleB.hs-boot----------------------------
module ModuleB where
  
-- import ModuleA (A)

foo :: Int
#---------------------------------------------------------

```

hie reports error when working with moduleA, and stops to provide diagnostics:

> 
> /private/var/folders/j8/1wn_n3g12171_ly1q141dr28507kn_/T/ghc-mod32137/ModuleA32136-3.hs:3:23: error:
>     Can't find /private/var/folders/j8/1wn_n3g12171_ly1q141dr28507kn_/T/ghc-mod32137/ModuleB32136-2.hs-boot

<img width="842" alt="Screen Shot 2019-03-14 at 13 37 53pm" src="https://user-images.githubusercontent.com/686076/54361196-711b5e80-465e-11e9-96c8-ece5238e9861.png">

hie version:
`Version 0.6.0.0, Git revision 726c37e52ac25e034ec25118eb1bea4dedb03aba (2457 commits) x86_64 ghc-8.6.3`
OS: Mac OS Sierra

**Comment by alanz** [comment]
@mpickering , does your new `hie-bios` deal with this properly?

**Comment by mpickering** [comment]
Yes it does but not perfectly. GHC has a lot of hs-boot files so I needed to deal with it.

**Comment by alanz** [comment]
Ok, so I think it is best to use this as a requirement on hie-bios, rather than try to fix it with the current setup.
